### PR TITLE
[Backport 2.x] [Search Pipelines] Add stats for search pipelines

### PR DESCRIPTION
**Pull request by mingshl**
### Description
manual back-porting https://github.com/opensearch-project/OpenSearch/pull/8053 and resolve conflict

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/6723

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by mingshl** [comment]
Need help in rerunning gradle check

**Comment by msfroh** [comment]
It looks like the Gradle check is failing because of the 1.3.11 release -- the buildBwcLinuxTar target is now building 1.3.12?

**Comment by msfroh** [comment]
Note that to avoid breaking backward compatibility tests on `main`, we will need to update the version checks on `main` to also check for versions on or after 2.9.0. Please create the PR for that and link it from this PR. They basically need to go in together (until we come up with a better solution).

See https://github.com/opensearch-project/OpenSearch/issues/8070 for discussion.

**Review by msfroh — CHANGES_REQUESTED** [review]

**Comment by mingshl** [comment]
> Note that to avoid breaking backward compatibility tests on `main`, we will need to update the version checks on `main` to also check for versions on or after 2.9.0. Please create the PR for that and link it from this PR. They basically need to go in together (until we come up with a better solution).
> 
> See #8070 for discussion.

createdhttps://github.com/opensearch-project/OpenSearch/pull/8416 to update the version checks on `main`

File: server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java [hunk 1]
```diff
@@ -202,6 +206,11 @@ public NodeStats(StreamInput in) throws IOException {
         } else {
             taskCancellationStats = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_2_9_0)) { // TODO Update to 2_9_0 when we backport to 2.x
```
[the remark this answers is not in the document]
Comment (reta) [reply server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java:209]: https://github.com/opensearch-project/OpenSearch/pull/8435 - forwardport to main

File: server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java [hunk 2]
```diff
@@ -430,6 +446,9 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_2_9_0)) {
             out.writeOptionalWriteable(taskCancellationStats);
         }
+        if (out.getVersion().onOrAfter(Version.V_2_9_0)) { // TODO: Update to 2_9_0 once we backport to 2.x
```
[the remark this answers is not in the document]
Comment (mingshl) [reply server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java:449]: The TODO will remove in 2.x once we merge in the version change to 2.9 https://github.com/opensearch-project/OpenSearch/pull/8416/files#diff-afe8d7298c117ffe707f8755d7e981f9f1103dbd0b040b40642efb555428aeecL196

**Review by tlfeng — APPROVED** [review]

**Comment by mingshl** [comment]
Thank you @tlfeng !

**Pull request merged**